src/utils.js

```diff
@@ -39,19 +39,19 @@ export const shallowClone = (obj, path = '') => {
   }
 };
 
-export const setByPath = (obj = {}, path = [], value) => {
+const HEAD = 'HEAD';
+export const setByPath = (obj, path = [], value) => {
```

Comment:
performance concern: should we check the `obj[path] === value` before calling all the following stuff? If no value changes, you hopefully don't have to anything.